gips/data/landsat/landsat.py

```diff
@@ -1346,3 +1396,187 @@ def _readraw(self, asset_type):
+                warp_base_band=os.path.join(tmpdir, warp_base_band_filename),
+                warp_data_type=' '.join((['2'] * len(warp_bands_bin))),
+                warp_nsample=warp_base_band_img.XSize(),
+                warp_nline=warp_base_band_img.YSize(),
+                warp_pixel_size=warp_pixel_size,
+                warp_upper_left_x=warp_base_band_img.MinXY().x(),
+                warp_upper_left_y=warp_base_band_img.MaxXY().y(),
+                warp_utm=self.utm_zone(),
+                out_bands=' '.join([os.path.join(tmpdir, basename(band) + '_warped.bin') for band in warp_bands_bin]),
+                out_base_band=os.path.join(tmpdir, basename(warp_base_band_filename)) + '_warped.bin',
+                out_pixel_size=out_pixel_size,
+                log_file='{}/cp_log.txt'.format(tmpdir),
+            )
+
+            parameter_file = os.path.join(tmpdir, 'parameter_file.inp')
+            with open(parameter_file, 'w') as param_file:
+                param_file.write(parameters)
+
+            shutil.copyfile(
+                os.path.join(os.path.dirname(__file__), 'lndortho.cps_par.ini'),
+                os.path.join(tmpdir, 'lndortho.cps_par.ini')
+            )
+
+            subprocess.check_call(["ortho", "-r", parameter_file])
+            
+            with open('{}/cp_log.txt'.format(tmpdir), 'r') as log:
+                xcoef_re = re.compile(r"x' += +([\d\-\.]+) +\+ +[\d\-\.]+ +\* +x +\+ +[\d\-\.]+ +\* y")
+                ycoef_re = re.compile(r"y' += +([\d\-\.]+) +\+ +[\d\-\.]+ +\* +x +\+ +[\d\-\.]+ +\* y")
+
+                for line in log:
+                    x_match = xcoef_re.match(line)
+                    if x_match:
+                        xcoef = float(x_match.group(1))
+                    y_match = ycoef_re.match(line)
+                    if y_match:
+                        ycoef = float(y_match.group(1))
+
+            x_shift = ((base_band_img.MinXY().x() - warp_base_band_img.MinXY().x()) / out_pixel_size - xcoef) * out_pixel_size
+            y_shift = ((base_band_img.MaxXY().y() - warp_base_band_img.MaxXY().y()) / out_pixel_size + ycoef) * out_pixel_size
+
+            with open('{}/{}_{}_coreg_args.txt'.format(self.path, self.id, datetime.strftime(self.date, "%Y%j")), 'w') as coreg_args:
+                coreg_args.write("x: {}\n".format(x_shift))
+                coreg_args.write("y: {}".format(y_shift))
+
+    def utm_zone(self):
+        """
+        Parse UTM zone out of `gdalinfo` output.
+        """
+        if getattr(self, 'utm_zone_number', None):
+            return self.utm_zone_number
+
+        asset = self.assets['C1']
+        any_band = [band for band in asset.datafiles() if band.endswith("TIF")][0]
+        ps = subprocess.Popen(["gdalinfo", "/vsitar/" + asset.filename + "/" + any_band], stdout=subprocess.PIPE)
+        ps.wait()
+        info = ps.stdout.read()
+        utm_zone_re = re.compile(".+UTM[ _][Z|z]one[ _](\d{2})[N|S].+", flags=re.DOTALL)
```

Comment:
Optional:  Still some `|` chars that could come out.  Should be a quick edit if you want.